src/large_pages/node_large_page.cc

```diff
@@ -359,12 +353,19 @@ MoveTextRegionToLargePages(const text_region& r) {
 #endif
 
   if (mprotect(start, size, PROT_READ | PROT_EXEC) == -1) goto fail;
-  // We need not `munmap(tmem, size)` in the above `OnScopeLeave` on success.
+
+  // We need not `munmap(tmem, size)` on success.
   tmem = nullptr;
-  return 0;
+  goto done;
 fail:
   PrintSystemError(errno);
-  return -1;
+  status = -1;
+done:
+  if (nmem != nullptr && nmem != MAP_FAILED && munmap(nmem, size) == -1)
```

Comment:
nit: maybe possible to avoid code repetition here what do you think ?

Reply:
Oh, right, with a `(nmem != nullptr && ... ) || (tmem != nullptr && ... )`. @jasnell had suggested that previously.

Reply:
Well, actually, you wanna `munmap` them both, even if the first one fails. So, you may `PrintSystemError()` because the first munmap failed, because the second munmap failed, twice because they both failed, or not at all.

Reply:
AFAICT to avoid this code repetition we have to go all the way to https://github.com/nodejs/node/pull/32570 which I feel is a bit too much.

Reply:
We could do `__attribute__((cleanup("DeleteMmapPointer")))` for `nmem` and `tmem` but I'm not sure how portable that is and we would have two levels of indirection because we'd have to store `nmem` along with `size` in some structure which would be passed to the deleter. Same for `tmem`.

Again, it seems too much complication for avoiding a single repetition.